packages/big-design/src/components/FileUploader/File.tsx

```diff
@@ -0,0 +1,56 @@
+import { DeleteIcon, DraftIcon } from '@bigcommerce/big-design-icons';
+import React, { HTMLAttributes, useMemo } from 'react';
+
+import { Button } from '../Button';
+
+import { FileStyled, ImageStyled, TextEllipsed } from './styled';
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  name: string;
+  isValid: boolean;
+  isSingle: boolean;
+  previewSrc: string | null;
+  onRemove(name: string): void;
+}
+
+export const File = ({ name, isValid, isSingle, previewSrc, onRemove, ...props }: Props) => {
+  const renderedFilePreview = useMemo(() => {
+    if (previewSrc) {
+      return <ImageStyled alt="preview" src={previewSrc} />;
+    }
+
+    return (
+      <DraftIcon
+        aria-hidden={false}
+        aria-label="file icon"
+        color="secondary50"
+        role="img"
+        size={40}
+      />
+    );
+  }, [previewSrc]);
+
+  return (
+    <FileStyled
+      {...props}
+      alignItems="center"
+      isValid={isValid}
+      justifyContent="space-between"
+      marginBottom={isSingle ? 'none' : 'small'}
+      paddingHorizontal="small"
+      role={isSingle ? undefined : 'listitem'}
```

Comment:
I think parent component should be responsible for role choosing. Component by it self is not a list item and doesn't know anything about place where it will be rendered.

Reply:
Done